moler/cmd/unix/socat.py

```diff
@@ -0,0 +1,148 @@
+        if is_full_line:
+            try:
+                self._command_failure(line)
+                self._parse_info_msg(line)
+                self._parse_output(line)
+            except ParsingDone:
+                pass
+        return super(Socat, self).on_new_line(line, is_full_line)
+
+    _re_error = re.compile(r'.* socat\[\d*\]\s[E|F]\s(?P<ERROR_MSG>.*)')
+
+    def _command_failure(self, line):
+        if self._regex_helper.search_compiled(Socat._re_error, line):
+            self.set_exception(CommandFailure(self, "ERROR: {}".format(self._regex_helper.group("ERROR_MSG"))))
+            raise ParsingDone
+
+    _re_info = re.compile(r'.* socat\[\d*\]\s(?P<INFO_MSG>[NWID].*)')
+
+    def _parse_info_msg(self, line):
+        if self._regex_helper.search_compiled(Socat._re_info, line):
+            info_msg = self._regex_helper.group("INFO_MSG")
+            self.current_ret['INFO'].append(info_msg)
+            raise ParsingDone
+
+    def _parse_output(self, line):
+        self.current_ret['RESULT'].append(line.strip())
+        raise ParsingDone
+
+
+COMMAND_OUTPUT_info_output = """
+xyz@debian:~$ socat -d -d pty,raw,echo=0,b9600 pty,raw,echo=0,b9600
+2016/01/16 12:57:51 socat[18255] N PTY is /dev/pts/2
+2016/01/16 12:57:51 socat[18255] N PTY is /dev/pts/4
+2016/01/16 12:57:51 socat[18255] N starting data transfer loop with FDs [5,5] and [7,7]
+xyz@debian:~$"""
+
+COMMAND_KWARGS_info_output = {
+    'options': '-d -d',
+    'input_options': 'pty,raw,echo=0,b9600',
+    'output_options': 'pty,raw,echo=0,b9600'
+}
+
+COMMAND_RESULT_info_output = {
+    'RESULT': [],
+    'INFO': ['N PTY is /dev/pts/2', 'N PTY is /dev/pts/4', 'N starting data transfer loop with FDs [5,5] and [7,7]']
+}
+
+
+COMMAND_OUTPUT_basic_output = """
+xyz@debian:~$ socat SYSTEM:date -
+Mon Aug 20 10:37:14 CEST 2018
+xyz@debian:~$"""
+
+COMMAND_KWARGS_basic_output = {
+    'input_options': 'SYSTEM:date',
+    'output_options': '-'
+}
+
+COMMAND_RESULT_basic_output = {
+    'RESULT': ['Mon Aug 20 10:37:14 CEST 2018'],
+    'INFO': []
+}
+
+
+COMMAND_OUTPUT_both_types = """
+xyz@debian:~$ socat -d - EXEC:'ssh -l userName server1.nixcraft.net.in',pty,setsid,ctty
+2018/08/20 11:14:46 socat[31539] W open("/dev/tty", O_NOCTTY, 0640): No such device or address
+ssh: Could not resolve hostname server1.nixcraft.net.in: Name or service not known
+2018/08/20 11:14:46 socat[31538] W waitpid(): child 31539 exited with status 255
+xyz@debian:~$"""
+
+COMMAND_KWARGS_both_types = {
+    'options': '-d',
+    'input_options': '-',
+    'output_options': "EXEC:'ssh -l userName server1.nixcraft.net.in',pty,setsid,ctty"
+}
+
+COMMAND_RESULT_both_types = {
+    'RESULT': ['ssh: Could not resolve hostname server1.nixcraft.net.in: Name or service not known'],
+    'INFO': ['W open("/dev/tty", O_NOCTTY, 0640): No such device or address',
+             'W waitpid(): child 31539 exited with status 255']
+}
+
+
+COMMAND_OUTPUT_version = """
+xyz@debian:~$ socat -V
```

Comment:
Please add parsing for that output, i.e. dict with `WITH_STDIO ` as key and `1` as value